lib/src/local_working_copy.rs

```diff
@@ -1752,6 +1762,18 @@ impl LockedWorkingCopy for LockedLocalWorkingCopy {
         Ok(())
     }
 
+    fn reset_to_empty(&mut self) -> Result<(), ResetError> {
+        self.wc
+            .tree_state_mut()
+            .map_err(|err| ResetError::Other {
+                message: "Failed to read the working copy state".to_string(),
+                err: err.into(),
+            })?
```

Comment:
We now have 5 copies (or very close) of this code. It's probably time to extract a method for it. Feel free to leave for a separate PR.

Reply:
OK, I'll leave for a separate PR.